spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -10,10 +18,32 @@
     end
 
     context "when sending a welcome_thread notification" do
-      xit "generates the appropriate broadcast to be sent to a user"
-      xit "it sends a welcome notification for that broadcast"
-      xit "it does not send duplicate welcome notification for that broadcast"
-      xit "does not send a notification to a user who has commented in a welcome thread"
+      let(:welcome_thread_article) { create(:article, title: "Welcome Thread") }
+      let(:welcome_thread_comment) { create(:comment, commentable: welcome_thread_article, user: user) }
+      let(:article) { create(:article) }
+      let(:comment) { create(:comment, commentable: article, user: receiving_user) }
+
+      it "generates the correct broadcast type and sends the notification to the user", :aggregate_failures do
+        expect(receiving_user.notifications.count).to eq(0)
+        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+
+        expect(receiving_user.notifications.count).to eq(1)
+        expect(receiving_user.notifications.first.notifiable).to eq(welcome_broadcast)
+      end
+
+      it "does not send a notification to a user who has commented in a welcome thread", elasticsearch: true do
+        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+
+        expect(user.notifications).to be_empty
```

Comment:
This test isn't doing what I'd expect. I think what it should be doing is:
1) creating a comment on a welcome thread for the user in a `before` block
2) calling the generator in the test
3) asserting the notifications for that user have not changed (since they shouldn't receive a welcome thread notification if they've already commented on the welcome thread).

Reply:
I have revamped this test since you've looked at it and it should be written and working as expected now! 🤞